model/jpa/src/main/java/org/keycloak/connections/jpa/EntityManagerFactoryProvider.java

```diff
@@ -0,0 +1,8 @@
+package org.keycloak.connections.jpa;
+
+import javax.persistence.EntityManagerFactory;
+
+public interface EntityManagerFactoryProvider {
```

Comment:
Introducing yet another "connection provider" is not the correct approach here. Looking at this PR a lot of changes/refactoring are done to the WF distribution, while if this wasn't introduced and the "copy" of DefaultJpaConnectionProviderFactory kept the WF distribution would have been more or less untouched by these changes and it would have been a lot easier and safer to review/merge this PR

Reply:
The changes do improve WF dist too. Test runs seem to be OK.

Replicating them is not ideal, but I understand your point about first trying this change in Quarkus, isolated. I'll close this PR and have both JPA and Liquibase providers duplicated.

It is kind of easy to check the impact of these changes in WF dist if you want too, just run the server using these changes.